### PR TITLE
RA-1008 changed loginPageController logic

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
https://issues.openmrs.org/browse/RA-1008

### CODE REVIEW

**Comment by AdamGrzybkowski** [comment]
This PR depend on [this one](https://github.com/openmrs/openmrs-module-uiframework/pull/40)

**Comment by dkayiwa** [comment]
@AdamGrzybkowski do you think you can look into these merge conflicts?

**Comment by AdamGrzybkowski** [comment]
@dkayiwa I've resolved conflicts, but could't replicate missing param issue.

**Comment by dkayiwa** [comment]
I fixed that in https://github.com/openmrs/openmrs-module-uiframework/commit/3605b5f7113f1e90dd69deb58a2f1cd915eb4f2f

**Comment by dkayiwa** [comment]
@AdamGrzybkowski this does not seem to meet all the ticket acceptance criteria. Are you looking into that?

**Comment by AdamGrzybkowski** [comment]
Yes. I will give you answer on JIRA once I'm done testing ;)

File: omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java [hunk 1]
```diff
@@ -120,33 +114,58 @@ public String get(PageModel model,
...
+                    String urlPath = url.getFile();
+                    String urlContextPath = urlPath.substring(0, urlPath.indexOf('/', 1));
+                    if (StringUtils.equals(pageRequest.getRequest().getContextPath(), urlContextPath)) {
+                        return true;
+                    }
+                }
+                catch (MalformedURLException e) {
+                    log.error(e.getMessage());
+                }
+			} else if(redirectUrl.startsWith("/openmrs")){
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java]: What happens when one is using a webapp context name other than openmrs? Something like bahmni?

File: omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java [hunk 2]
```diff
@@ -120,33 +114,58 @@ public String get(PageModel model,
...
-					catch (MalformedURLException e) {
-						log.error(e.getMessage());
-					}
-				} else {
-					redirectUrl = pageRequest.getRequest().getHeader("Referer");
-				}
+	private boolean isUrlWithinOpenmrs(PageRequest pageRequest, String redirectUrl){
+		if (StringUtils.isNotBlank(redirectUrl)) {
+			if (redirectUrl.startsWith("http://") || redirectUrl.startsWith("https://")) {
+				try {
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java]: Can you improve the formatting here?

**Pull request merged**